src/client/src/rt-components/tear-off/ExternalWindow.tsx

```diff
@@ -42,18 +43,19 @@ const ExternalWindow: FC<ExternalWindowProps> = ({
     const getWindow = async () => {
       externalWindow = await platform.window.open(config, release)
       if (externalWindow) {
-        window.addEventListener('beforeunload', release)
```

Comment:
please note, were listening for close events on a wrong window - main instead of the one that has just been opened